mvrx/src/main/kotlin/com/airbnb/mvrx/ScriptableMvRxStateStore.kt

```diff
@@ -1,22 +1,29 @@
 package com.airbnb.mvrx
 
-import io.reactivex.Observable
-import io.reactivex.subjects.BehaviorSubject
+import kotlinx.coroutines.channels.BroadcastChannel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.consumeEach
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.buffer
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.flow
 
 /**
  * A [MvRxStateStore] which ignores standard calls to [set]. Instead it can be scripted via calls to
  * [next]. This is intended to be used for tests only, and in particular UI tests where you wish to test
  * how your UI code reacts to different ViewModel states. This is not as useful for unit testing your view model,
  * as business logic in state reducers will not be used.
  */
+@Suppress("EXPERIMENTAL_API_USAGE")
 class ScriptableMvRxStateStore<S : Any>(initialState: S) : MvRxStateStore<S> {
 
-    private val subject: BehaviorSubject<S> = BehaviorSubject.createDefault(initialState)
-
-    override val observable: Observable<S> = subject.distinctUntilChanged()
-
-    override val state: S
-        get() = subject.value!!
+    private val stateChannel = BroadcastChannel<S>(capacity = Channel.BUFFERED)
```

Comment:
@gpeal I think you can achieve the [`BehaviorSubject`](http://reactivex.io/RxJava/javadoc/io/reactivex/subjects/BehaviorSubject.html) behavior using [`ConflatedBroadcastChannel`](https://stackoverflow.com/questions/57345311/how-can-i-send-items-to-a-kotlin-flow-like-a-behaviorsubject).

The [`BroadcastChannel`](https://stackoverflow.com/questions/57345311/how-can-i-send-items-to-a-kotlin-flow-like-a-behaviorsubject) was used to achieve the [`PublishSubject`](http://reactivex.io/RxJava/javadoc/io/reactivex/subjects/PublishSubject.html).

Reply:
@jpventura `ConflatedBroadcastChannel` conflates inputs. It's actually possible to set the state faster than it can be delivered to consumers, even with a buffer. I was able to demonstrate that in some of the added tests. Tricky stuff ;)